scipy/sparse/_matrix.py

```diff
@@ -52,31 +49,9 @@ def __rmul__(self, other):
 
     # Restore matrix power
     def __pow__(self, other):
-        M, N = self.shape
-        if M != N:
-            raise TypeError('sparse matrix is not square')
-
-        if isintlike(other):
-            other = int(other)
-            if other < 0:
-                raise ValueError('exponent must be >= 0')
-
-            if other == 0:
-                from ._construct import eye
-                return eye(M, dtype=self.dtype)
-
-            if other == 1:
-                return self.copy()
-
-            tmp = self.__pow__(other // 2)
-            if other % 2:
-                return self @ tmp @ tmp
-            else:
-                return tmp @ tmp
-
-        if isscalarlike(other):
-            raise ValueError('exponent must be an integer')
-        return NotImplemented
+        from .linalg import matrix_power
```

Comment:
Since SciPy has switched to lazy imports, this can be a higher level import now.